src/lib.rs

```diff
@@ -515,12 +515,25 @@ impl Group {
 /// assert_eq!(faces, ["😀", "😃", "😄", "😁", "😆"]);
 /// ```
 #[inline]
-pub fn iter() -> impl Iterator<Item = &'static Emoji> {
+pub fn iter() -> impl Iterator<Item = &'static Emoji> + Clone {
     crate::gen::EMOJIS
         .iter()
         .filter(|emoji| matches!(emoji.skin_tone(), Some(SkinTone::Default) | None))
 }
 
+/// Returns an iterator over all shortcodes.
+///
+/// # Examples
+///
+/// ```
+/// let shortcodes: Vec<_> = emojis::shortcodes().take(3).collect();
+/// assert_eq!(shortcodes, ["algeria", "pouting_face", "waning_crescent_moon"]);
+/// ```
+#[inline]
+pub fn shortcodes() -> impl Iterator<Item = &'static str> + Clone {
+    crate::gen::shortcode::MAP.keys().into_iter().copied()
```

Comment:
This will include duplicates, is that the intention? E.g. `:dolphin:` and `:flipper:` both refer to the dolphin emoji.

**Alternatives:**

This includes only the first shortcode, and the order is more stable.
```rust
pub fn shortcodes() -> impl Iterator<Item = &'static str> + Clone {
    crate::gen::EMOJIS
        .iter()
        .filter_map(|emoji| emoji.shortcode())
}
```

This includes all shortcodes and the order is more stable.
```rust
pub fn shortcodes() -> impl Iterator<Item = &'static str> + Clone {
    crate::gen::EMOJIS
        .iter()
        .flat_map(|emoji| emoji.shortcodes())
}
```

Reply:
The intention is to be able to see all possible shortcodes to pick one based on a substring. The stability of the order is technically not important. The flatmap approach actually works quite well considering that there are 1898 emojis and 1913 shortcodes in total. I was concerned about performance but since the numbers are almost the same it does not matter.